Add support for a list of paths in each PluginType

Gee19 opened the pull request:
Refs #SDE-3585

aeldamaty commented:
+1 looks good to me

hanifvirani commented on yantra/manager.py:
```diff
@@ -45,21 +45,22 @@ def _get_modules(self):
 
-        # walk the plugins folder
-        for root, subdir, files in os.walk(self.plugin_type.path):
+        for p in self.plugin_type.paths:
```
Rename `p` to `path`

hanifvirani commented on yantra/manager.py:
```diff
@@ -45,21 +45,22 @@ def _get_modules(self):
-        for root, subdir, files in os.walk(self.plugin_type.path):
+        for p in self.plugin_type.paths:
+            # walk the plugins folder
+            for root, subdir, files in os.walk(p):
```
This method is getting a bit ridiculous at this point. Might be good to extract the walking part into another method that takes a plugin.

Gee19 commented:
I removed the existing _get_modules method because I felt it was an unnecessary abstraction after I extracted the folder walking to it's own method(populate_modules). I am now calling populate_modules for each path inside the get_plugins method. 

Also should I return modules from the populate_modules method or leave it the way it is? (Mutating the object passed in)

Let me know what you think @hanifvirani @aeldamaty

aeldamaty commented:
+1. If @hanifvirani has any preferences for way the method should act that's fine. Both sound good to me.

hanifvirani commented:
Yeah, I would leave the `get_plugins` method the way it was and just refactor `_get_modules` into two pieces.

Gee19 commented:
@hanifvirani Done. I'm still unhappy with populate_plugin_modules_from_path mutating the list being passed to it rather than just returning a module and _get_modules appending to plugin_modules.

hanifvirani commented:
As discussed in person, turns out we don't need this.
